### DISCUSSION TITLE
Preparing macOS installation running for a few hours

### DISCUSSION THREAD

**Discussion by saby-getsorbet** [grahampugh/erase-install#239] [Q&A]
Hi all,
I am testing the script on one of the macOS (11.6),
the cache download was successful,
running the upgrade with --depnotify and the process in on the updating macOS - preparing installer for a few hours now.
is it normal ? 
how can I check what is stuck ?

**Comment by grahampugh** [comment]
You could check the log file, but often times it's just `startosinstall` getting stuck and there is no output - nothing I can help with there, `erase-install.sh` just a wrapper script for built-in tools.

**Reply by saby-getsorbet** [reply]
Thanks,
how can I stop this process on the device -
![d403b693-4ff6-44c6-a6c8-6cacfdc43697](https://user-images.githubusercontent.com/95706401/167586451-780eda21-92c0-44a8-a6f8-7592d424bec7.jpg)

**Comment by grahampugh** [comment]
Default DEPNotify QuitKey applies - Cmd + Ctrl + x
https://github.com/jamf/DEPNotify#quit-1

**Comment by saby-getsorbet** [comment]
Thanks,
Terminating the process and I see this screen which ask me for permission where the install is located.
anyway I can bypass this ?
![image](https://user-images.githubusercontent.com/95706401/167677738-acf1ab46-0eab-47e9-8b5c-5b473143bbc4.png)

**Reply by grahampugh** [reply]
Yes. don't use Documents, Downloads or Desktop. 
I provide a package installer for the script which installs `erase-install.sh`and dependencies into `/Library/Management/erase-install`. That's the recommended installation method, but if you're doing it differently, choose somewhere not restricted like `/Users/Shared` or the folder I already mentioned.

**Reply by saby-getsorbet** [reply]
I didn't change the default  location,
can I set it to upgrade at next reboot ?

**Reply by grahampugh** [reply]
If you download the script directly, then default location is the folder you run the script from. You choose where to put it. 

But if you install the package it puts it in an appropriate location. 

There is no option to upgrade at next reboot. This script is just a wrapper around startosinstall, and there is no such option available in that tool.

**Reply by saby-getsorbet** [reply]
Thanks,
will try several option with the script and see what will the best for us.
